pulp_smash/tests/docker/cli/test_copy.py

```diff
@@ -0,0 +1,244 @@
+# coding=utf-8
+"""Tests for copying docker units between repositories."""
+from __future__ import unicode_literals
+import re
+
+import unittest2
+from packaging.version import Version
+
+from pulp_smash import utils
+from pulp_smash.tests.docker.cli import utils as docker_utils
+
+
+class CopyAllImagesTestCase(docker_utils.BaseTestCase,
+                            docker_utils.SuccessMixin):
+    """Test copying all Images from one repository to another."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create and sync a docker repository with a v1 registry."""
+        super(CopyAllImagesTestCase, cls).setUpClass()
+        if cls.cfg.version < Version('2.8'):
+            raise unittest2.SkipTest('These tests require Pulp 2.8 or above.')
+        # Create and sync a repository with a v1 feed to bring some Images in.
+        docker_utils.create_repo(
+            cls.cfg, cls.repo_id, upstream_name='library/busybox',
+            sync_v1=True, sync_v2=False)
+        docker_utils.sync_repo(cls.cfg, cls.repo_id)
+
+        # Now create a feedless repository and copy the images from the first
+        # repository over.
+        cls.copy_target = utils.uuid4()
+        docker_utils.create_repo(cls.cfg, cls.copy_target)
+        cls.completed_proc = docker_utils.copy(cls.cfg, 'image', cls.repo_id,
+                                               cls.copy_target)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the copy_target repo."""
+        super(CopyAllImagesTestCase, cls).tearDownClass()
+        docker_utils.delete_repo(cls.cfg, cls.copy_target)
+
+    def test_positive_copy_output(self):
```

Comment:
In just about all of the tests in Pulp Smash, communications with Pulp occur in `setUpClass`, and assertions occur in test methods. However, this test is structured so that communications with Pulp occur in test methods. I don't think that's ideal. Can we push these communications into `setUpClass` and make the test methods concern themselves just with assertions?